src/components/VeggyBody/VeggyProductItem/ProductItem.jsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from "react";
+import './ProductItem.css';
+
+export default function ProductItem(props) {
+    const {
+        product,
+        selectedProducts,
+        setSelectedProducts,
+        zoomProductImage,
+        setShakingCart,
+    } = props;
+
+    const [count, setCount] = useState(1);
+    const [buttonText, setButtonText] = useState("ADD TO CART");
+
+    function handleChangeButtonText() {
+        setButtonText("ADDED");
+
+        setTimeout(() => {
+            setButtonText("ADD TO CART");
+        }, 1000);
+    }
+
+    const incrementProductQuantity = () => {
+        setCount(n => n + 1);
+    };
+
+    const decrementProductQuantity = () => {
+        if (count === 1) {
+            return;
+        }
+
+        setCount(count - 1);
+    };
+
+    function handleSetProductQuantity(e) {
+        setCount(e.target.value);
+    }
+
+    function addProduct() {
```

Comment:
давай перейменуємо, щоб одразу було очевидно куди ми добавляємо продукт. Більше конкретики
```suggestion
    function addToCart() {
```